be/src/runtime/stream_load/stream_load_pipe.cpp

```diff
@@ -180,7 +180,7 @@ Status StreamLoadPipe::no_block_read(uint8_t* data, size_t* data_size, bool* eof
                         write_buf->put_bytes((char*)data, bytes_read);
                         write_buf->flip();
                         // error happens iff pipe is cancelled
-                        RETURN_IF_ERROR(_push_front(write_buf));
+                        RETURN_IF_ERROR(_no_lock_push_front(write_buf));
```

Comment:
How about setting it as _read_buf?

Reply:
`_read_buf` is only used in no_block_read(uint8_t* data, size_t* data_size, bool* eof) and  read(uint8_t* data, size_t* data_size, bool* eof). It is not used StreamLoadPipe::read() and StreamLoadPipe::no_block_read().
If we set the read_buf here, next time when no_block_read(), it can not find the buf in buf_queue.